Update api key and backend basic auth docs (#5111)
wso2/docs-apim: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-key-authentication.md

Follow the instructions below to use API Key Authentication with WSO2 Choreo Connect:

- If you are using API Manager to deploy the API, follow [API Key Authentication with WSO2 API Manager]({{base_path}}/design/api-security/api-authentication/secure-apis-using-api-keys).

> - If you are using APICTL to deploy the API, follow [API Key Authentication using OpenAPI security schemes](#api-key-authentication-using-openapi-security-schemes).
> 

Reviewer: ```suggestion
- If you are using APICTL to deploy the API, follow [API Key Authentication for Standalone Mode](#api-key-authentication-for-standalone-mode).

```

> ## API Key Authentication using OpenAPI security schemes

Reviewer: ```suggestion
## API Key Authentication for Standalone Mode
```

> Follow the instructions below to enable API Key Authentication using OpenAPI security schemes.

Reviewer: ```suggestion
Follow the instructions below to enable API Key Authentication for standalone mode using OpenAPI security schemes.
```

!!! info
    When you enable API authentication for an API and deploy via WSO2 API Manager, it will override the API Key Authentication defined in the OpenAPI definition.

>     Therefore, the following steps work for Choreo Connect when the API is not deployed via WSO2 API Manager. 
>     So use the following steps when you want to enable API Key authentication when deploying an API using the APICTL.

Reviewer: ```suggestion
    Therefore, the following steps are only for Choreo Connect standalone mode and would not work when the API is deployed via WSO2 API Manager. Hence, use the following steps when you want to enable API Key authentication when deploying an API using APICTL.
```

### Step 1  - Define the API Key security scheme

Security schemes must be defined on the OpenAPI definition under `securitySchemes`. One or more API Key security schemes can be used simultaneously, similar to a logical OR operator. Define the API Key security scheme as follows:

| Field Name | Description   |
| -----------| --------------|

> | `name`     | Specify a unique name. This name will be used to refer to the scheme on API level or resource level. |

Reviewer: ```suggestion
| `name`     | Name of the header or query|
```
